lib/matplotlib/backends/registry.py

```diff
@@ -82,11 +280,130 @@ def list_builtin(self, filter_=None):
             Backend names.
         """
         if filter_ == BackendFilter.INTERACTIVE:
-            return self._BUILTIN_INTERACTIVE
+            return [k for k, v in self._BUILTIN_BACKEND_TO_GUI_FRAMEWORK.items()
+                    if v != "headless"]
         elif filter_ == BackendFilter.NON_INTERACTIVE:
-            return self._BUILTIN_NOT_INTERACTIVE
+            return [k for k, v in self._BUILTIN_BACKEND_TO_GUI_FRAMEWORK.items()
+                    if v == "headless"]
+
+        return [*self._BUILTIN_BACKEND_TO_GUI_FRAMEWORK]
+
+    def list_gui_frameworks(self):
+        """
+        Return list of GUI frameworks used by Matplotlib backends.
+
+        Returns
+        -------
+        list of str
+            GUI framework names.
+        """
+        return [k for k in self._GUI_FRAMEWORK_TO_BACKEND if k != "headless"]
+
+    def load_backend_module(self, backend):
+        """
+        Load and return the module containing the specified backend.
+
+        Parameters
+        ----------
+        backend : str
+            Name of backend to load.
+
+        Returns
+        -------
+        Module
+            Module containing backend.
+        """
+        module_name = self._backend_module_name(backend)
+        return importlib.import_module(module_name)
+
+    def resolve_backend(self, backend):
+        """
+        Return the backend and GUI framework for the specified backend name.
+
+        If the GUI framework is not yet known then it will be determined by loading the
+        backend module and checking the ``FigureCanvas.required_interactive_framework``
+        attribute.
+
+        This function only loads entry points if they have not already been loaded and
+        the backend is not built-in and not of ``module://some.backend`` format.
+
+        Parameters
+        ----------
+        backend : str or None
+            Name of backend, or None to use the default backend.
+
+        Returns
+        -------
+        Tuple of backend (str) and GUI framework (str or None).
```

Comment:
Also should be type only:
```suggestion
        tuple of (str, str or None)
```

Reply:
Actually, I just remembered that for numpydoc, this probably shouldn't be documented as returning a tuple, but as 2 separate things: https://numpydoc.readthedocs.io/en/latest/format.html#returns

So something like:
```
        Returns
        -------
        backend : str
            The backend name.
        framework : str or None
            The framework, which will be None for a backend that is non-interactive.
```
And the same for the other method that returns a tuple.

Reply:
Thanks for this. I've done as you suggested, which also removes the "headless" references that aren't needed.